### PR TITLE
Add initial BSD support to the compiler

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
There are only a handful (probably less than 10) of places in the compiler and JitBuilder that deal with LINUX or OSX.  OSX was the most recent port to a Unix-like operating system.  Perhaps you can grep through the compiler and jitbuilder source for the `OSX` macro and consider whether you need to add support there as well for BSD.  Or have you done that already?

**Comment by dnakamura** [comment]
@0xdaryl  Updated. The only other thing was fixing some strcmp code. I can't  guarantee  I've got everything, but with this everything compiles and the compiler/jitbuilder tests pass (there are still a few port library / ras bugs I'm hunting down)

File: compiler/ras/IgnoreLocale.hpp [hunk 1]
```diff
@@ -30,41 +30,11 @@ int toupper_ignore_locale(int c);
 int stricmp_ignore_locale(const char *s1, const char *s2);
 int strnicmp_ignore_locale(const char *s1, const char *s2, size_t n);
 
-#if (J9ZOS390 || AIXPPC || LINUX || OSX)
```
Comment (0xdaryl) [inline compiler/ras/IgnoreLocale.hpp:33]: Apologies for not seeing the connection, but what does removing all this code have to do with enabling BSD support?

Comment (dnakamura) [reply compiler/ras/IgnoreLocale.hpp:33]: The code provides a case insenstive string compare. Ignoring the dead `PILOT` code path, we have the windows case, the main case which covers zos, aix, linux and osx (which uses `strcasecmp`), and the fallback case which provides a simplistic implementation. There a sighned/unsigned warnings in the fallback code (which by default trigger errors). Given that our main case uses `strcasecmp`, which is in the POSIX spec its pretty safe to assume we will have it everywhere (except for windows which we already cover)

Comment (0xdaryl) [reply compiler/ras/IgnoreLocale.hpp:33]: OK.  So it is related to the BSD support then as opposed to just some cleanup work.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**